src/components/home/HomePage.js

```diff
@@ -1,41 +1,71 @@
-import React, { Component } from 'react';
+import React, { useState, useEffect } from 'react';
 import './HomePage.css';
 
-class HomePage extends Component {
-  render() {
-    return (
-      <div className="HomePage">
-        <div className="uk-grid">
-          <div className="uk-width-1-5@s" style={{ background: '#0e0b0bb3', color: 'white' }}>
-            <div className="uk-offcanvas-content">
-              <button
-                className="uk-button uk-button-default uk-margin-small-right"
-                type="button"
-                data-uk-toggle="target: #offcanvas-push"
-              >
-                Push
-              </button>
+function HomePage() {
+  const [data, setData] = useState(null);
 
-              <div id="offcanvas-push" data-uk-offcanvas="mode: push; overlay: true">
-                <div className="uk-offcanvas-bar">
-                  <button className="uk-offcanvas-close" type="button" data-uk-close="true"></button>
+  useEffect(() => {
+    fetch('https://cors-anywhere.herokuapp.com/https://api.github.com/repos/iiitv/algos/contents', {
+      method: 'GET',
+      credentials: 'same-origin',
+      headers: new Headers({
+        'Access-Control-Allow-Origin': '*',
+      }),
+      mode: 'cors',
+    })
+      .then((res) => res.json())
+      .then((data) => {
+        // console.log(data);
+        let val = data.map((item) => {
```

Comment:
this is a use case of `Array.filter`, also use `const`.
Also, I feel we should not keep DOM elements in the state, rather keep API response in state and write a loop in render

Reply:
okkk..sir i will fix it.